### DISCUSSION TITLE
Tag numeric values in text fields that scale with recipe scaling

### DISCUSSION THREAD

**Discussion by SpencerDub** [mealie-recipes/mealie#4749] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Certain quantities or numbers in a recipe do not scale when the recipe scales, because they are not ingredient quantities. For example:

1. A note on an ingredient: "1 cup chopped shallots, **about 4 shallots**"
2. A recipe step might instruct you to use a pseudo-ingredient that is created during cooking: "Reserve **1/2 cup** of the cooking water"
3. A recipe step might instruct you to use part of a listed ingredient: "Add **1 Tbsp.** of the chili powder"

### Please provide a concise description of the feature that would resolve your issue.

If certain numeric strings within other text fields could be denoted as scalable, perhaps with a syntax as simple as `{{value}}`:

> 1 cup chopped shallots, about {{4}} shallots

> Reserve {{1/2}} cup of the cooking water

> Add {{1}} Tbsp. of the chili powder

Then, when the recipe is scaled, these values are scaled just like ingredient quantities.

### Please consider and list out some caveats or tradeoffs made in your design decision

This would probably be an advanced user function, since it feels "code-y" in a way that might scare off users without technical knowledge.

Would have to account for instances when the contents of the tag are not a numeric value—what does Mealie do when told that `{{1 cup}}` (rather than `{{1}} cup`) is supposed to scale?

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by SpencerDub** [comment]
Having just looked at my Mealie instance a little more, it seems the current Servings field handles ignoring the non-numeric characters the way I would want for the `{{1 cup}}` situation.

**Comment by peavyprowler** [comment]
That's exactly what I'd like to have!
Would be great to have this!

**Comment by ChipSkylark37** [comment]
+1 as well.

Especially when scaling, but even in general, I'd like to be able to insert the ingredients into the steps (more than just linking them) so it reads as:

"Add (1) cup sugar, and beat (2 eggs),etc" 

instead of 

"Add sugar, and beat eggs"

LINKED INGREDIENT:
1 cup sugar
2 eggs

I've also mentioned on some other discussions baking temperature. If we had that as a sort of 'linked ingredient' it could be good to have as a way to insert that into steps. I know I like to reference it a few times, but if I forget to change one, it could mess things up. My oven is a bit 'off' so if I have in step 1. "Preheat oven to (xxx degrees)" and then in step 5 is "Place try into oven, which should now be at (xxx degrees)", to have that linked and entered would be awesome.

**Reply by peavyprowler** [reply]
Hey @ChipSkylark37,
what do you mean with "linked ingredients"? Is that a feature I haven't discovered yet or just some text you add for yourself?

**Reply by ChipSkylark37** [reply]
When in edit mode, go to one of the steps and click the three dots:

![image](https://github.com/user-attachments/assets/96d1c2c8-3ea8-4e14-84df-d19ad24fbb2a)

Then you can click to link ingredients, per step:

![image](https://github.com/user-attachments/assets/a0e1392b-f650-4628-b6da-52e75fc87c7f)

![image](https://github.com/user-attachments/assets/da8a1238-2177-4e8b-972a-e177039e1e5a)

You can also link the same ingredient to multiple steps if desired (to reference them more than once).

There's some cool things in that menu, take a look and try them out.

**Reply by peavyprowler** [reply]
Wow, totally overlooked this! Thanks for showing. 
Indeed, this helps a lot, but the inline referencing would be even more awesome.
